flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/ExpressionReducer.scala

```diff
@@ -68,7 +68,7 @@ class ExpressionReducer(
 
       // Skip expressions that contain python functions because it's quite expensive to
       // call Python UDFs during optimization phase. They will be optimized during the runtime.
-      case (_, e) if PythonUtil.containsFunctionOf(e, FunctionLanguage.PYTHON) =>
+      case (_, e) if PythonUtil.containsFunctionOf(e, findPythonFunction = true) =>
```

Comment:
Rename the method? `containsFunctionOf` and the doc is not very accurate anymore. Maybe `isPythonFunction`? The flag `findPythonFunction` is not needed anymore.

Reply:
Thanks a lot for the suggestion. The contains is used here because this method will also check the inputs of this RexNode, i.e., whether the rexCall tree contains python or non-python functions, so maybe contains is better than is?

As for the `isPythonFunction` flag, do you mean that we can remove the flag and use two methods? one is `containsPythonFunction()`, the other is `containsNonPythonFunction()`?

Reply:
I prefer `containsPythonFunction()` and `containsNonPythonFunction`, because it's more clear than `findPythonFunction = false` flag. Meanwhile, I think `recursive` is also not a very clear flag. So I would suggest following utils:

```
boolean containsPythonCall(RexNode)
boolean containsNonPythonCall(RexNode)
boolean isPythonCall(RexNode)
boolean isNonPythonCall(RexNode)
```

This also follows `RexUtil.isConstant(RexNode)` and `RexUtil.containsFieldAccess(RexNode)`

Reply:
+1

Reply:
Thank you for your suggestions.